### PR TITLE
chore: add sample android-kotlin app

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
## Description

-  In this PR, we removed the existing `android-java` sample app from the `tests/e2e` directory.
- Created a new `android-kotlin` app with the RudderStack Android SDK setup.
- Added a README file with instructions for setting up the Android project.
- Added RudderTyper events in the sample app.

**NOTE**: I've replaced all the credentials with the dummy values.

Screenshot:
image

# Thanks for taking the time to contribute to RudderTyper!

It is highly appreciated that you take the time to help improve RudderTyper.
We appreciate it if you would take the time to document your Pull Request.

Sadly, if we don't receive enough information, or the Pull Request doesn't
align well with our roadmap, we might respectfully thank you for your time, and close the issue.

## Respect earns Respect 👏

Please respect our [Code of Conduct](CODE_OF_CONDUCT.md), in short:

* Using welcoming and inclusive language.
* Being respectful of differing viewpoints and experiences.
* Gracefully accepting constructive criticism.
* Focusing on what is best for the community.
* Showing empathy towards other community members.

### CODE REVIEW

File: tests/e2e/android-kotlin/readme.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- Rename `rudderstack.properties.sample` to `rudderstack.properties`.
+- Update the `writeKey`, `dataPlaneUrl`, and `controlPlaneUrl` with your specific values.
+
+### Step 3: Sync Gradle
+
+- Sync the Gradle files by clicking on the `Sync Now` link in the top right corner of the Android Studio window. Or, you can click on the `File` menu and select `Sync Project with Gradle Files`:
+
+    ![syncGradle](images/syncGradle.png)
+
+### Step 4: Add Generated Files
```
Comment (saikumarrs) [inline tests/e2e/android-kotlin/readme.md:23]: Can we add some automated script that can build the RudderTyper and add the output to this sample app?

File: tests/e2e/android-kotlin/readme.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- Crete a new package `ruddertyper` under `app/src/main/java/com/rudderstack/typewriterexample`:
+    ![createNewRudderTyperPackage.png](images/createNewRudderTyperPackage.png)
+- Place the RudderTyper-generated files in the `ruddertyper` package.
+
+### Step 5: Implement Event Tracking
```
Comment (saikumarrs) [inline tests/e2e/android-kotlin/readme.md:28]: Some basic sample event tracking should be automatically part of the app. Can we add it if not already?

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: tests/e2e/android-kotlin/readme.md (hunk 1 above)
Comment (saikumarrs) [reply tests/e2e/android-kotlin/readme.md:23]: Yes, use @akashrpo's help to get it working.
The sample apps should be ready to test with minimal effort.

File: tests/e2e/android-kotlin/readme.md (hunk 2 above)
Comment (1abhishekpandey) [reply tests/e2e/android-kotlin/readme.md:28]: Adding sample events requires the RudderTyper-generated files to be present in the sample app, which is not currently included. Therefore, I’ve provided instructions in the README as well as [here](https://github.com/rudderlabs/rudder-typer/blob/9f8835cb5200641ec614b5d7f0af2d8c05e7e314/tests/e2e/android-kotlin/app/src/main/java/com/rudderstack/typewriterexample/MainActivity.kt#L55).

File: tests/e2e/android-kotlin/readme.md (hunk 1 above)
Comment (akashrpo) [reply tests/e2e/android-kotlin/readme.md:23]: `tests/e2e/android-java/typewriter.yml` <- This is deleted, but we can have `ruddertyper.yml` here and we just need to run the existing scripts `npx ruddertyper prod or dev` to get the files in the desired location as per the path value in yml file.

File: tests/e2e/android-kotlin/readme.md (hunk 2 above)
Comment (akashrpo) [reply tests/e2e/android-kotlin/readme.md:28]: The sample events can get autogenerated from 
`tests/e2e/android-java/app/src/main/java/com/segment/generated/plan.json` <- deleted file
we can have a sample schema here with sample naming. <- the `npx ruddertyper prod/dev` first looks for schema in plan.json if this file is not there then only it fetches from API.

File: tests/e2e/android-kotlin/readme.md (hunk 1 above)
Comment (1abhishekpandey) [reply tests/e2e/android-kotlin/readme.md:23]: I've added the relevant steps to generate the corresponding ruddertyper files. Please go through the updated readme.
I've also added the output to the sample app.

PS: I've replaced all the credentials with dummy values.

File: tests/e2e/android-kotlin/readme.md (hunk 2 above)
Comment (saikumarrs) [reply tests/e2e/android-kotlin/readme.md:28]: @1abhishekpandey, is this also done now with your recent changes?

Comment (1abhishekpandey) [reply tests/e2e/android-kotlin/readme.md:28]: (I was going to comment.)
I've added the sample events in the app, please find it [here](https://github.com/rudderlabs/rudder-typer/blob/799b84e7439362c1b26570d86e2a71eca5907aef/tests/e2e/android-kotlin/app/src/main/java/com/rudderstack/ruddertyper/MainActivity.kt#L57).

**Review by saikumarrs — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**